p2p/messenger.go

```diff
@@ -0,0 +1,342 @@
+package p2p
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/centrifuge/go-centrifuge/errors"
+
+	"github.com/libp2p/go-libp2p-host"
+	"github.com/libp2p/go-libp2p-protocol"
+
+	pb "github.com/centrifuge/go-centrifuge/protobufs/gen/go/protocol"
+	ggio "github.com/gogo/protobuf/io"
```

Comment:
Let's stick one protobuf implementation else this will cause unneeded compatibility issues later